File: TFG/ftp_tfg/lib/domain/usecases/get_sync_history.dart
import '../entities/sync_record.dart';
import '../repositories/ftp_repository.dart';
import '../interfaces/i_get_sync_history_use_case.dart';

class GetSyncHistory implements IGetSyncHistoryUseCase {
  final FtpRepository repository;

  GetSyncHistory(this.repository);

  @override
  Future<List<SyncRecord>> execute(String ownerId) =>
      repository.getSyncHistory(ownerId);
}

File: TFG/ftp_tfg/backend/src/application/use-cases/sync/get-sync-history.ts
import type { SyncRecord } from '../../../domain/entities/sync-record.js';
import type { MonitoringRepository } from '../../../domain/repositories/monitoring-repository.js';

export class GetSyncHistory {
  constructor(private readonly monitoringRepository: MonitoringRepository) {}

  execute(ownerId: string): Promise<SyncRecord[]> {
    return this.monitoringRepository.getSyncHistory(ownerId);
  }
}

